index.js

```diff
@@ -1,38 +1,89 @@
+//final week 10
 import express from 'express'
 import bodyParser from 'body-parser'
 import redis from 'redis'
 
-// make the one and only express object     -   done?
 const app = express()
 
-// make a database connection to redis      -   done?
 var urlencodedParser = bodyParser.urlencoded({ extended: false})
 const db = redis.createClient({
     url: 'redis://localhost:6379'
 })
-await db.connect()     // if this errors, make sure redis is running    -   done?
+await db.connect()    
 
-// add routes
-// this route serves static files
 app.use(express.static('public'))
 
-// this route serves the home page, using code
-app.post('/add', urlencodedParser, async (req, res) => {
-    await db.set("form-data", req.body.answerText)
-    res.send(`<!DOCTYPE html><html><body>Data saved to Redis key named 'form-data'</body></html>`)
-    res.redirect()
+
+
+app.get('/', async(req, res) => {
+    res.send(`<!DOCTYPE html><html><body><A href="/home.html">Say Hello</A></body></html>`)
 })
+
+app.post('/add', urlencodedParser, async(req, res) => {
+    var quizName = req.body.quizName
+    await db.set('quizName', quizName)
+    console.log(quizName)
+    await db.hSet('questionCount', quizName, '0')
+    const g = await db.get('quizName')
+    console.log(`g is ${g}`)
+    await db.set('ttttt', 1)
+    
+    res.send(`<!DOCTYPE html>
+    <html>
+    <head>
+        <title>Add a Question</title>
+    </head>
+    <body>
+        <h1>Add Your First Question</h1>
+            <h2>Here is Your Quiz's Name:</h2>
+                ${quizName}
+            <form action="/add/${quizName}/1" method="post">
+                <h2>Type Your Question Below:</h2>
+                <textarea id="questionText" name="questionText" rows="5" cols="40"></textarea>
+                <h2>Type the Answer Below:</h2>
+                <input type="text" id="answerText" name="answerText"><br />
+                <br />
+                <button type="submit">Submit This Question</button>
+            </form>
+    </body>
+    </html>`)
+})
+
 app.post('/add/:quizName/:count', urlencodedParser, async (req, res) => {
-    await db.set("form-data", req.body.mytext)
-    res.send(`<!DOCTYPE html><html><body>Data saved to Redis key named 'form-data'</body></html>`)
-    res.redirect()
+    const quizName = await db.get('quizName')
+    console.log(`The name of this quiz is: ${quizName}`)
+    await db.incr('ttttt')
```

Comment:
The quizName and count come from the URL, not from redis.

See `req.params` in the debugger.